desc/magnetic_fields.py

```diff
@@ -1803,6 +1810,299 @@ def from_surface(
         )
 
 
+class OmnigenousField(Optimizable, IOAble):
+    """Model field with perfect omnigenity.
+
+    Notes
+    -----
+    Doesn't conform to MagneticField API, as it only knows about :math:`|B|` in flux
+    coordinates, not vector B in lab coordinates.
+
+    """
+
+    _io_attrs_ = [
+        "_L_well",
+        "_M_well",
+        "_L_shift",
+        "_M_shift",
+        "_N_shift",
+        "_NFP",
+        "_helicity",
+        "_well_basis",
+        "_shift_basis",
+        "_B_lm",
+        "_x_lmn",
+    ]
+
+    def __init__(
+        self,
+        L_well=0,
+        M_well=2,
+        L_shift=0,
+        M_shift=0,
+        N_shift=0,
+        NFP=1,
+        helicity=(1, 0),
+        B_lm=None,
+        x_lmn=None,
+    ):
+        self._L_well = int(L_well)
+        self._M_well = int(M_well)
+        self._L_shift = int(L_shift)
+        self._M_shift = int(M_shift)
+        self._N_shift = int(N_shift)
+        self._NFP = int(NFP)
+        self.helicity = helicity
+        self._well_basis = ChebyshevPolynomial(L=self.L_well)
+        self._shift_basis = ChebyshevDoubleFourierBasis(
+            L=self.L_shift,
+            M=self.M_shift,
+            N=self.N_shift,
+            NFP=self.NFP,
+            sym="cos(t)",
+        )
+        if B_lm is None:
+            self._B_lm = np.concatenate(
+                (
+                    np.ones((self.M_well,)),
```

Comment:
In docstring, specify what exactly this is (the first M_well are the spline eta knot values right? and the next L_well*M_well are the coefficients of the chebyshev polynomial in rho corresponding to each of the spline knots)

Reply:
see updated docstring